File: Python/td3.py
# ### Auteur : MOHAMED BOUHARTI
# ## TD 3

# ## Uncomment specific exercise


print("-----------------------------------")

# print("\n------Exercice ??------\n")
# if num < 10:
#     uniter = num
#     print(f"Unite : {num}")
# elif num < 100:
#     dizaine = num // 10
#     uniter = num % 10
#     print(f"dizaine: {dizaine}, Unite : {uniter}")
# elif num < 1000:
#     cenain = num // 100
#     dizaine = (num // 10) % 10
#     uniter = num % 10
#     print(f"cenain : {cenain}, dizaine: {dizaine}, uniter: {uniter}")
# else:
#     print("Invalid number!")

# print("\n------Exercice 1------\n")

# nbr = int(input("Enter a number: "))
# k = 1
# while k<=nbr:
#     print(k)
#     k+=1

# print("\n------Exercice 2------\n")

# A = 0
# B = int(input("Enter B:"))
# C = int(input("Enter C:"))

# if B == C:
#     A+=1

# while C>=0:
#     C = int(input("Enter C:"))
#     if B == C:
#         A+=1
# print("A = ",A)

# print("\n------Exercice 3------\n")

# i=0
# g_num = int(input("Entre un nomber: "))
# n=i+1
# i+=1

# while i<6:
#     num = int(input("Entre un nomber: "))

#     if num > g_num:
#         g_num = num
#         n=i+1

#     i+=1

# print("Grand nomber est: ",g_num)
# print("Sa position est: ",n)

# import math

# print("\n------Exercice 4------\n")

# nbr = int(input("Entre un nomber: "))
# S=1
# i=2
# # while i < math.sqrt(nbr):
# while (i*i) <= nbr:

#     if (nbr%i)==0:
#         S+=i
#         if (nbr//i)!=i:
#             S+=nbr//i
    
#     i+=1
# print(S)
# if S==nbr and nbr!=1:
#     print("'",nbr,"' est parfait.")
# else :
#     print("'",nbr,"' n'est pas parfait.")

# print("\n------Exercice 4------\n")

# num = int(input("Entre une nomber: "))

# som = 0

# for i in range(1,num):

#     if num%i==0:
#         som+=i
    
# if som==num:
#     print(f"Le nomber '{num}' est parfait.")
# else:
#     print(f"Le nomber '{num}' n'est pas parfait.")


# print("\n------Exercice 5------\n")

# num = int(input("Entre un nomber: "))

# while num<10 or num>20:

#     if num<10:
#         print("Plus Petit!")
#     if num>20:
#         print("Plus Grand!")

#     num = int(input("Entre un nomber: "))


# print("\n------Exercice 6------\n")
# num = int(input("Entre une nomber: "))

# stop = num+10
# #while num < stop:
# #    num += 1
# #    print(num)

# for i in range(0,10):
#    num += 1
#    print(num)

# print("\n------Exercice 7------\n")

# num=int(input("Entre un nomber: "))
# g_num=num
# i=0
# n=i+1
# i+=1

# while num!=0:
#     num=int(input("Entre un nomber: "))
#     if num>g_num:
#         g_num=num
#         n=i+1
#     i+=1

# print("Grand nomber: ",g_num)
# print("Position: ",n)

# print("\n------Exercice 8------\n")

# n = 0
# ch = str(input("Entre un nomber ([non] ou [Non] pour sortie): "))

# while ch != "non" and ch != "Non":
#     n += 1
#     ch = str(input("Entre un nomber ([non] ou [Non] pour sortie): "))
    
# print("Le nomber de valeurs saisire est :",n)

# print("\n------Exercice 9------\n")

# n = 0
# p = 0
# imp = 0

# nbr = int(input("Entre un nomber n (n<0 pour sortie): "))

# while nbr >= 0:
#     n+=1
#     if nbr%2 ==0:
#         p+=1
#     else:
#         imp+=1

#     nbr = int(input("Entre un nomber n (n<0 pour sortie): "))
    
# print("pair: ",p)
# print("impair: ",imp)
# print("total numbers: ",n)

# print("\n------Exercice 10------\n")

# c=input("Entre une charactaire: ")

# ch=str("")

# while c!='.':
    
#     if c == '0':
#         ch+=' '
#     else:
#         ch+=str(c)
    
#     c=input("Entre une charactaire: ")
    
# print(ch)

# print("\n------Exercice 9------\n")

# pair=0
# impair=0
# counter=0

# while True:

#     num = int(input("Etre un nomber:"))
    
#     if num<0:
#         break
    
#     counter+=1

#     if num %2==0:
#         pair+=1
#     else:
#         impair+=1



# print("Nombre de pairs :", pair)
# print("Nombre d'impairs :", impair)
# print("Total :", counter)

# print("\n------Exercice 10------\n")

# phrase=str("")

# while True:
    
#     ch=input("Entre une caractere: ")
    
#     if ch==".":
#         break
#     elif ch=="0":
#         phrase+=" "
#     else:
#         phrase+=ch

# print(f"phrase: {phrase}")

File: C/td3.c
/*
Auteur : MOHAMED BOUHARTI
TD 3

Uncomment specific exercise
*/

#include <stdio.h>

int main()
{
    // printf("\n------Exercice 1------\n");
    // int k = 1;
    // int nbr;

    // printf("Entre un nombner: ");
    // scanf("%d", &nbr);

    // while(k<=nbr)
    // {
    //     printf("%d\n",k);
    //     k+=1;
    // }
    // do
    // {
    //     printf("%d\n", k);
    //     k += 1;
    // } while (k <= nbr);

    // printf("\n------Exercice 2------\n");
    // int A = 0;
    // int B = 0;
    // int C = 0;

    // printf("Entre B: ");
    // scanf("%d", &B);
    // printf("Entre C: ");
    // scanf("%d", &C);

    // if (B == C)
    // {
    //     A += 1;
    // }

    // while (C >= 0)
    // {
    //     printf("Entre C: ");
    //     scanf("%d", &C);

    //     if (B == C)
    //     {
    //         A += 1;
    //     }
    // }

    // printf("Entre A = %d ",A);

    // printf("\n------Exercice 3------\n");

    // int i = 0;
    // int n = 0;
    // int num = 0;
    // int g_num = 0;

    // printf("Entre le nomber numero %d:", i + 1);
    // scanf("%d", &g_num);
    // n = i + 1;
    // i+=1;

    // do
    // {
    //     printf("Entre le nomber numero %d:", i + 1);
    //     scanf("%d", &num);

    //     if (num > g_num)
    //     {
    //         g_num = num;
    //         n = i + 1;
    //     }

    //     i += 1;
    // } while (i < 6);

    // printf("Grand nomber: %d\n",g_num);
    // printf("Sont position: %d\n",n);

    // printf("\n------Exercice 4------\n");

    // int nbr = 0;

    // printf("Entre un nomber (entier): ");
    // scanf("%d", &nbr);

    // int S = 0;
    // for (int i = 1; i * i <= nbr; ++i)
    // {
    //     if ((nbr % i) == 0)
    //     {
    //         S += i;

    //         if (((nbr / i) != i)&&(nbr / i)!=nbr && i != nbr)
    //         {
    //             S += (nbr / i);
    //         }
    //     }
    // }

    // if (S == nbr)
    // {
    //     printf("%d est parfait.\n", nbr);
    // }
    // else
    // {
    //     printf("%d n'est pas parfait.\n", nbr);
    // }

    // printf("\n------Exercice 5------\n");

    // int num = 0;

    // do
    // {
    //     printf("Entre un nomber: ");
    //     scanf("%d", &num);

    //     if (num < 10)
    //     {
    //         printf("Plus petit!\n");
    //     }
    //     if (num > 20)
    //     {
    //         printf("Plus Grand!\n");
    //     }

    // } while (num < 10 || num > 20);

    // printf("\n------Exercice 6------\n");

    // int num = 0;

    // printf("Entre un nomber: ");
    // scanf("%d", &num);

    // int stop = num + 10;

    // // while (num < stop)
    // // {
    // //     num+=1;
    // //     printf("%d\n",num);
    // // }

    // for (int i = 0; i < 10; i++)
    // {
    //     num+=1;
    //     printf("%d\n",num);
    // }

    // printf("\n------Exercice 7------\n");

    // int i=0;
    // int g_num=0;
    // printf("Entre un nomber ([0] pour sortie) %d :",i+1);
    // scanf("%d",&g_num);
    // int n=i+1;
    // i+=1;

    // int num;
    // do{
    //     printf("Entre un nomber ([0] pour sortie) %d :",i+1);
    //     scanf("%d",&num);
    //     if(num>g_num){
    //         g_num=num;
    //         n=i+1;
    //     }
    //     i+=1;
    // }while(num!=0);

    // printf("Grand nomber: %d\n",g_num);
    // printf("Sont position: %d\n",n);

    // #include <string.h>

    //     printf("\n------Exercice 8------\n");

    //     int n = 0;
    //     char ch[10];

    //     printf("Entre un nomber ([non] ou [Non] pour sortie): ");
    //     scanf("%9s", ch);

    //     while (strcmp(ch, "non") != 0 && strcmp(ch, "Non") != 0)
    //     {
    //         printf("ch = %s\n",ch);
    //         n += 1;
    //         int c;
    //         while((c=getchar()!='\n')&&c!=EOF);

    //         printf("Entre un nomber ([non] ou [Non] pour sortie): ");
    //         scanf("%9s", ch);
    //     }

    //     printf("Le nomber de valeurs sasis est: %d\n", n);

    // printf("\n------Exercice 9------\n");

    // int num = 0;
    // int p = 0;
    // int imp = 0;
    // int n;

    // do
    // {
    //     printf("Entre un nomber ([0] pour sortie): ");
    //     scanf("%d", &num);

    //     if (num != 0)
    //     {
    //         if (num % 2 == 0)
    //         {
    //             p += 1;
    //         }
    //         else if (num % 2 != 0)
    //         {
    //             imp += 1;
    //         }
    //         n += 1;
    //     }

    // } while (num != 0);

    // printf("Pair: %d\n",p);
    // printf("Impair: %d\n",imp);
    // printf("nomber: %d\n",n);

    // #include <string.h>

    //     printf("\n------Exercice 10------\n");

    // char ch[100];
    // int i = 0;
    // // char c[2];
    // char c1;

    // do
    // {
    //     printf("ch = %s\n", ch);

    //     printf("Entre une charactaire: ");
    //     scanf(" %c", &c1);

    //     if (c1 != '\n')
    //     {
    //         if (c1 == '0')
    //         {
    //             strcat(ch, " ");
    //         }
    //         else
    //         {
    //             ch[i] = c1;
    //         }
    //         i++;
    //     }

    //     // Method 2
    //     // printf("Entre une charactaire: ");
    //     // scanf("%1s", c);
    //     // if (c[0] == '0')
    //     // {
    //     //     strcat(ch, " ");
    //     // }
    //     // else
    //     // {
    //     //     strcat(ch, c);
    //     // }
    // } while (i < 100);

    return 0;
}
